server/src/main/java/com/objectcomputing/checkins/services/request_notifications/CheckServicesImpl.java

```diff
@@ -1,52 +1,33 @@
 
 @Singleton
 public class CheckServicesImpl implements CheckServices {
-    private final FeedbackRequestRepository feedbackReqRepository;
-    public static final String FEEDBACK_REQUEST_NOTIFICATION_SUBJECT = "check-ins.application.feedback.notifications.subject";
-    public static final String FEEDBACK_REQUEST_NOTIFICATION_CONTENT = "check-ins.application.feedback.notifications.content";
-    public static final String submitURL = "https://checkins.objectcomputing.com/feedback/submit?requestId=";
-    private EmailSender emailSender;
-    private String notificationSubject;
-    private String notificationContent;
-    private final MemberProfileServices memberProfileServices;
+    private final FeedbackRequestServicesImpl feedbackRequestServices;
+    private final FeedbackRequestRepository feedbackRequestRepository;
 
-    public CheckServicesImpl(FeedbackRequestRepository feedbackReqRepository,
-                             @Named(MailJetConfig.HTML_FORMAT) EmailSender emailSender,
-                             @Property(name = FEEDBACK_REQUEST_NOTIFICATION_SUBJECT) String notificationSubject,
-                             @Property(name = FEEDBACK_REQUEST_NOTIFICATION_CONTENT) String notificationContent,
-                             MemberProfileServices memberProfileServices) {
-        this.feedbackReqRepository = feedbackReqRepository;
-        this.emailSender = emailSender;
-        this.notificationContent = notificationContent;
-        this.notificationSubject = notificationSubject;
-        this.memberProfileServices = memberProfileServices;
+    public CheckServicesImpl(FeedbackRequestServicesImpl feedbackRequestServices,
+                             FeedbackRequestRepository feedbackRequestRepository) {
+        this.feedbackRequestServices = feedbackRequestServices;
+        this.feedbackRequestRepository = feedbackRequestRepository;
     }
 
     @Override
-    public boolean GetTodaysRequests() {
+    public boolean sendScheduledEmails() {
         LocalDate today = LocalDate.now();
-        List<FeedbackRequest> todaysRequests = new ArrayList<>();
-        todaysRequests.addAll(feedbackReqRepository.findByValues(null, null, null, today, null, null));
-        for (FeedbackRequest req: todaysRequests) {
-            MemberProfile from = memberProfileServices.getById(req.getCreatorId());
-            String fromName  = from.getFirstName() + " " + from.getLastName();
-            String newContent =  notificationContent + "<a href=\""+submitURL+req.getId()+"\">Check-Ins application</a>.";
-            emailSender.sendEmail(fromName, from.getWorkEmail(), notificationSubject, newContent);
+        List<FeedbackRequest> feedbackRequests = feedbackRequestRepository.findBySendDateBeforeAndStatusEqual(today, "pending");
+        for (FeedbackRequest req: feedbackRequests) {
+            feedbackRequestServices.sendNewRequestEmail(req);
+            req.setStatus("sent");
+            feedbackRequestRepository.update(req);
```

Comment:
Here, it might make more sense to update the feedback request first, and then if the update succeeds, send the email. I'm just thinking we would want to keep these in sync in case either of these calls fail.

Reply:
I think that the email is more likely to fail than the db call.